main.py

```diff
@@ -166,39 +202,80 @@ def add_system_metrics_tegra(args, messages):
                         meta={"name": name.lower()},
                     )
                 )
-        else:
-            logging.info("tegrastats did not return any data. skipping...")
 
-    except Exception:
-        logging.exception("failed to get tegra system metrics")
+        except Empty:
```

Comment:
You _could_ make the "except Empty" case break instead of continue and change the outer loop to just while True:.

In practice, it shouldn't make much of a different since you're doing a non-blocking .get anyway.